extensions/amp-consent/amp-consent.md

```diff
@@ -0,0 +1,220 @@
+    <button on="tap:consent-element.reject" role="button">Reject</button>
+    <button on="tap:consent-element.dismiss" role="button">Dismiss</button>
+  </div>
+</amp-consent>
+```
+AMP displays prompt UI on page load or by user interaction. The prompt UI is hidden with any of the three user actions described below.
+
+#### Prompt Actions
+
+There are three types of user actions that are associated with the consent prompt: `accept`, `reject` and `dismiss`.
+
+To enable the user to choose a consent state and hide the prompt UI, add an `on` attribute to a button with the
+following value scheme `on="event:idOfAmpConsentElement.accept/reject/dismiss"`
+
+* `accept`: publisher instructs AMP to remember the accept decision to the consent, unblocks components waiting for the consent, and hides the prompt UI.
+
+* `reject`: publisher instructs AMP to remember the reject decision to the consent, cancels `buildCallback` of components waiting for the consent, and hides the prompt UI.
+
+* `dismiss`: instruct AMP to cancel `buildCallback` of components waiting for the consent, and hides the prompt UI.
+
+
+### Post-prompt UI (optional)
+
+Post-prompt UI is defined with the `<amp-consent>` JSON configuration object. The `postPromptUI` refers to a child element of `<amp-consent>` by id.
+
+```html
+<amp-consent layout="nodisplay" id="consent-element">
+  <script type="application/json">
+  {
+    "consents": {
+      "consent-foo": {
+        ...
+        "promptUI": "consent-ui"
+      }
+    },
+    "postPromptUI": "post-consent-ui"
+  }
+  </script>
+  <div id="consent-ui">
+     ...
+  </div>
+  <div id="post-consent-ui">
+     <button on="tap:consent-element.dismiss" role="button">Dismiss</button>
+  </div>
+
+</amp-consent>
+```
+
+When defined, the post-prompt UI is shown when all prompt UIs have been hidden, or initially on page load if no prompt UI was triggered.
+
+#### Post-prompt action
+
+The post-prompt UI provides one user action type that can be used to allow the user to manage a previously set consent. Use `prompt` to display a prompt for a given consent instance. Add an `on` attribute to a button with the following value scheme `on="event:idOfAmpConsentElement.prompt"`.
+
+## Blocking behaviors
+
+The `<amp-consent>` element can be used to block any other AMP components on the page (except `<amp-consent>` itself).
+
+To block components, add the `data-block-on-consent` attribute to the AMP component. This ensures that `buildCallback` of the component isn't called until consent has been accepted, or if the consent prompt has been skipped by the `checkConsentHref` response when consent is unknown.
+
+AMP may support customizing blocking behaviors in the future. Because of this, the value of `data-block-on-consent` is reserved for now, please don't specify a value to the attribute.
+
+Individual components may override this behavior to provide more specialized handling. Please refer to each component's documentation for details.
```

Comment:
can you link this to the other md file you put together, probably on the words "specialized handling".

Reply:
I figure this is not the correct place to link to the other doc. We can talk about that later.